mirage/views/moderator.ts

```diff
@@ -1,9 +1,15 @@
-import { HandlerContext, Request, Schema } from 'ember-cli-mirage';
+import { HandlerContext, ModelInstance, Request, Schema } from 'ember-cli-mirage';
+import { MirageRegistrationProvider } from 'ember-osf-web/mirage/factories/registration-provider';
+import { MirageCollectionProvider } from 'ember-osf-web/mirage/factories/collection-provider';
 
 export function addModerator(this: HandlerContext, schema: Schema, request: Request) {
     const { parentID } = request.params;
     const attrs = this.normalizedRequestAttrs('moderator');
-    const provider = schema.registrationProviders.find(parentID);
+    let provider: ModelInstance<MirageRegistrationProvider | MirageCollectionProvider> =
+        schema.registrationProviders.find(parentID);
+    if (request.url.includes('collections')) {
+        provider = schema.collectionProviders.find(parentID);
+    }
```

Comment:
Can we do `instanceof ModelInstance<MirageRegistrationProvider>` and/or `instance of ModelInstance<MirageCollectionProvider>` here to determine which provider we are getting? Or we could look at the url. I just don't like the fact that we are trying to assign `provider` to `schema.registrationProviders.find(parentID)` then reassign to something else afterwards depending on the request url.